### PR TITLE
[tempo-distributed] Add host aliases

### PR DESCRIPTION
**Pull request by YuleZ**
I need to set hostAliases in my environment for loki-distributed.
This PR adds this feature to the other helm-charts as well as for loki-distributed where I need this change initially.

### CODE REVIEW

**Review by zalegrala — APPROVED** [review]
This looks good to me, thanks for putting this together.

**Comment by zalegrala** [comment]
The CI is failing with some details.  Let me know if you have a chance to review.

File: charts/tempo-distributed/values.yaml [hunk 1]
```diff
@@ -1880,6 +1935,11 @@ enterpriseGateway:
   # If you want to use your own proxy URLs, set this to false.
   useDefaultProxyURLs: true
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:1939]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 2]
```diff
@@ -1795,6 +1845,11 @@ tokengenJob:
 # Can only be enabled if enterprise.enabled is true - requires license.
 adminApi:
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:1849]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 3]
```diff
@@ -1782,6 +1827,11 @@ license:
 # enterprise.enabled is true - requires license.
 tokengenJob:
   enable: true
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:1831]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 4]
```diff
@@ -1506,6 +1546,11 @@ gateway:
   enabled: false
   # -- Number of replicas for the gateway
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:1550]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 5]
```diff
@@ -1331,6 +1366,11 @@ memcached:
 memcachedExporter:
   # -- Specifies whether the Memcached Exporter should be enabled
   enabled: false
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:1370]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 6]
```diff
@@ -785,6 +815,11 @@ enterpriseFederationFrontend:
   enabled: false
   # -- Number of replicas for the federation-frontend
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:819]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 7]
```diff
@@ -647,6 +672,11 @@ queryFrontend:
       backend: 127.0.0.1:3100
   # -- Number of replicas for the query-frontend
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:676]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 8]
```diff
@@ -517,6 +537,11 @@ compactor:
 querier:
   # -- Number of replicas for the querier
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:541]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 9]
```diff
@@ -446,6 +461,11 @@ distributor:
 compactor:
   # -- Number of replicas for the compactor
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:465]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 10]
```diff
@@ -341,6 +351,11 @@ metricsGenerator:
 distributor:
   # -- Number of replicas for the distributor
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:355]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 11]
```diff
@@ -224,6 +229,11 @@ metricsGenerator:
   annotations: {}
   # -- Number of replicas for the metrics-generator
   replicas: 1
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:233]:
```suggestion
  hostAliases: []
```

File: charts/tempo-distributed/values.yaml [hunk 12]
```diff
@@ -106,6 +106,11 @@ ingester:
   annotations: {}
   # -- Number of replicas for the ingester
   replicas: 3
+  # -- hostAliases to add
+  hostAliases: [ ]
```
Comment (zanhsieh) [inline charts/tempo-distributed/values.yaml:110]:
```suggestion
  hostAliases: []
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**